src/ray/raylet/scheduling/cluster_task_manager.cc

```diff
@@ -71,6 +73,11 @@ bool ClusterTaskManager::SchedulePendingTasks() {
       // blocking where a task which cannot be scheduled because
       // there are not enough available resources blocks other
       // tasks from being scheduled.
+      Priority task_priority = work_it->first.first;
+      if(task_priority >= block_requested_priority_){
```

Comment:
This loop will choose whether to keep the task for local execution and put the tasks in a "dispatch queue", for dispatching to a worker (`tasks_to_dispatch_`), so also need to make sure to block tasks in that queue from being dispatched.

You can do this in `DispatchScheduledTasksToWorkers`.